lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -219,9 +234,72 @@ defmodule Mix.Compilers.Elixir do
         do: relative
   end
 
-  defp each_file(source) do
-    Mix.shell.info "Compiled #{source}"
+  defp each_file(pid, verbose, source) do
+    if verbose, do: Mix.shell.info "Compiled #{source}"
+
+    Agent.get(pid, fn {_, sources} -> Enum.find(sources, & source(&1, :source) == source) end)
+    |> warn_for_missing_remote_functions()
+  end
+
+  defp warn_for_missing_remote_functions(source_record) do
+    compile_dispatches = source(source_record, :compile_dispatches)
+    runtime_dispatches = source(source_record, :runtime_dispatches)
+    source = source(source_record, :source)
+
+    do_warn_for_missing_remote_functions(compile_dispatches ++ runtime_dispatches, source)
+  end
+
+  defp do_warn_for_missing_remote_functions(runtime_dispatches, source) do
+    Enum.sort_by(runtime_dispatches, fn {_, _, line} -> line end)
+    |> Enum.each(fn {module, {func, arity}, line} ->
+      if Code.ensure_loaded?(module) do
+        unless function_exported?(module, func, arity) or
+               is_macro?(module, func, arity) or
```

Comment:
I think there's 2 more bugs here, first we should probably only check if it's a macro if it's in the compile dispatches, and also right now macro dispatches are appearing in runtime rather than compile as I think they should be.